pull/Java8TASK/files/Runner.java

```diff
@@ -0,0 +1,57 @@
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.List;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+public class Runner{
+
+    public static void main(String[] args) {
+//        1. Create an ArrayList implementation for 9 entities (3 – for a superclass and 2 – for every subclass).
+        Trial[] array = {new Trial("Maria Zaytseva", 85, 68),
+                new Trial("Victoria Volkova", 73, 87),
+                new Trial("Misha Muromskiy", 65, 98),
+                new LightTrial("Tanya Plachet", 74, 63),
+                new LightTrial("Sveta High", 45, 36),
+                new StrongTrial("Andrei Stain", 57, 89),
+                new StrongTrial("Sergei Gavron", 78, 78),
+                new ExtraTrial("Sasha Komar", 45, 45, 19),
+                new ExtraTrial("Denis Kovalev", 84, 75, 92)};
+        List<Trial> list = new ArrayList<>(Arrays.asList(array));
+
+//        2. Print the collection content (one element per line).
+        list.forEach(System.out::println);
+//        3. Print the number of passed trials.
+        System.out.println(list
+                            .stream()
+                            .filter(Trial::isPassed)
+                            .count());
+//      4. Sort the collection by the sum of first and second marks.
+        Function<Trial, Integer> sumOfMarks = trial -> (trial.mark1 + trial.mark2);
+        list.sort(Comparator.comparing(sumOfMarks));
+//      5. Print sums of first and second marks from the collection (one sum per line).
+        list.stream()
+                .map(sumOfMarks)
+                .forEach(System.out::println);
+//      6. Create a new collection from unpassed trials, clear all marks and print this collection. Check whether all trials are failed (the result type is boolean).
+        List<Trial> loosers = list.stream()
+                .filter(p -> !p.isPassed())
+                .map(Trial::clone)
+                .peek(Trial::clearAllMarks)
+                .peek(System.out::println)
+                .collect(Collectors.toList());
+
+        System.out.println("Is it true that all loosers don't pass exam??? -> " + loosers.stream().noneMatch(Trial::isPassed));
+//      7. Create a numeric array from sums of first and second marks of sorted collection (see item 4) and print it in the format:
+//        sum[0], sum[1], … , sum[sum.length - 1]
+        int[] sumArray = list.stream()
+                .mapToInt(sumOfMarks::apply)
+                .toArray();

```

Comment:
если 
`Function<Trial, Integer> sumOfMarks = trial -> (trial.mark1 + trial.mark2);`
то
```suggestion
        int[] sumArray = list.stream()
                .mapToInt(sumOfMarks)
                .toArray();
```